src/Sarif.Viewer.VisualStudio/CodeAnalysisResultManager.cs

```diff
@@ -672,36 +678,80 @@ private static string GetCommonSuffix(string firstPath, string secondPath)
         private void AttachToDocumentChanges(uint docCookie, IVsWindowFrame pFrame)
         {
             ThreadHelper.ThrowIfNotOnUIThread();
+
+            IVsTextView vsTextView = VsShellUtilities.GetTextView(pFrame);
+            if (vsTextView == null)
+            {
+                return;
+            }
+
+            if (vsTextView.GetBuffer(out IVsTextLines vsTextLines) != VSConstants.S_OK)
+            {
+                return;
+            }
+
+            IComponentModel componentModel = (IComponentModel)Package.GetGlobalService(typeof(SComponentModel));
+            if (componentModel == null)
+            {
+                return;
+            }
+
+            IVsEditorAdaptersFactoryService editorAdapterFactoryService = componentModel.GetService<IVsEditorAdaptersFactoryService>();
+            if (editorAdapterFactoryService == null)
+            {
+                return;
+            }
+
+            ITextBuffer textBuffer = editorAdapterFactoryService.GetDataBuffer(vsTextLines);
+
             string documentName = GetDocumentName(docCookie);
 
-            if (!string.IsNullOrEmpty(documentName))
+            if (string.IsNullOrEmpty(documentName))
+            {
+                return;
+            }
+
+            if (RunIndexToRunDataCache == null)
+            {
+                return;
+            }
+
```

Comment:
Move this check up. If we don't have the cache, then there is no sense doing all the rest of this.